waiter/src/waiter/process_request.clj

```diff
@@ -544,40 +526,49 @@
                         ; request-state-chan should be explicitly closed after the request finishes processing
                         request-state-chan (async/tap control-mult (au/latest-chan) false)
                         queue-timeout-ms (:queue-timeout-ms instance-request-properties)
-                        instance (fa/<? (prepare-instance instance-rpc-chan service-id reason-map start-new-service-fn request-state-chan
-                                                          queue-timeout-ms reservation-status-promise metric-group
-                                                          add-debug-header-into-response!))]
+                        instance-timer (metrics/service-timer service-id "get-available-instance")
+                        timed-instance (metrics/with-timer
+                                         instance-timer
+                                         (fa/<? (prepare-instance instance-rpc-chan service-id reason-map
+                                                                  start-new-service-fn request-state-chan queue-timeout-ms
+                                                                  reservation-status-promise metric-group)))
+                        instance (:out timed-instance)
+                        instance-elapsed (:elapsed timed-instance)]
+                    (statsd/histo! metric-group "get_instance" instance-elapsed)
                     (-> (try
                           (log/info "suggested instance:" (:id instance) (:host instance) (:port instance))
                           (confirm-live-connection-without-abort)
                           (let [endpoint (request->endpoint request waiter-headers)
-                                {:keys [error] :as response}
-                                (metrics/with-timer!
-                                  (metrics/service-timer service-id "backend-response")
-                                  (fn [nanos]
-                                    (add-debug-header-into-response! "X-Waiter-Backend-Response-ns" nanos)
-                                    (statsd/histo! metric-group "backend_response" nanos))
-                                  (async/<!
-                                    (make-request-fn instance request instance-request-properties
-                                                     passthrough-headers endpoint metric-group)))
+                                backend-timer (metrics/service-timer service-id "backend-response")
+                                timed-response (metrics/with-timer
+                                                 backend-timer
```

Comment:
We can inline `backend-timer` and avoid introducing a new variable

Reply:
Done.